scripts/check_compliance.py

```diff
@@ -314,11 +314,12 @@ def check_no_undef_outside_kconfig(self, kconf):
         # Kconfig files that are not part of the main Kconfig tree, which will
         # trigger false positives until we do something fancier. Skip
         # doc/releases too, which often references removed symbols.
-        grep_cmd = "git grep --only-matching --line-number -I --null " \
-                   "--extended-regexp --word-regexp CONFIG_[A-Z0-9_]+ " \
-                   "-- :!samples :!tests :!doc/releases"
+        grep_cmd = ["git", "grep", "--only-matching", "--line-number", "-I",
+                    "--null", "--perl-regexp",
+                    r"\bCONFIG_[A-Z0-9_]+\b(?!\s*#)",
```

Comment:
Is there any harm in making this `(?!\s*##)` by the way, just to make it a bit more specific?

The check searches comments and stuff too, where maybe `CONFIG_FOO  #...` could show up in some context, though it's unlikely.

Reply:
Done for both.  I also missed an instance that could be removed from the whitelist the first time through (`CONFIG_UART_`); I only checked for ones that matched the new exclusion (i.e. that where present as simple token pastes), but I managed to miss that one.

Reply:
LGTM